Userland/Libraries/LibJS/Runtime/SetPrototype.cpp

```diff
@@ -232,4 +233,85 @@ JS_DEFINE_NATIVE_FUNCTION(SetPrototype::union_)
     return result;
 }
 
+// 2 Set.prototype.intersection ( other ), https://tc39.es/proposal-set-methods/#sec-set.prototype.intersection
+JS_DEFINE_NATIVE_FUNCTION(SetPrototype::intersection)
+{
+    auto& realm = *vm.current_realm();
+
+    // 1. Let O be the this value.
+    // 2. Perform ? RequireInternalSlot(O, [[SetData]]).
+    auto* set = TRY(typed_this_object(vm));
+
+    // 3. Let otherRec be ? GetSetRecord(other).
+    auto other_record = TRY(get_set_record(vm, vm.argument(0)));
+
+    // 4. Let resultSetData be a new empty List.
+    auto* result = Set::create(realm);
+
+    // 5. Let thisSize be the number of elements in O.[[SetData]].
+    auto this_size = set->set_size();
+
+    // 6. If thisSize ≤ otherRec.[[Size]], then
+    if (this_size <= other_record.size) {
+        // a. For each element e of O.[[SetData]], do
+        for (auto& element : *set) {
+            // i. If e is not empty, then
+            //     1. Let inOther be ToBoolean(? Call(otherRec.[[Has]], otherRec.[[Set]], « e »)).
+            auto in_other = TRY(call(vm, *other_record.has, other_record.set, element.key)).to_boolean();
+            //     2. If inOther is true, then
+            if (in_other) {
+                // a. Append e to resultSetData.
+                result->set_add(element.key);
+            }
+        }
+    }
+
```

Comment:
no newline here is CxByte's preferred style, methinks

Reply:
adjusted :^)